crates/bevy_reflect/src/attributes.rs

```diff
@@ -150,6 +150,7 @@ macro_rules! impl_custom_attribute_methods {
             $self.custom_attributes().get::<T>()
         }
 
+        #[allow(rustdoc::redundant_explicit_links)]
         /// Gets a custom attribute by its [`TypeId`](std::any::TypeId).
         ///
         /// This is the dynamic equivalent of [`get_attribute`](Self::get_attribute).
```

Comment:
This is sometimes used with `TypeId` in scope and sometimes without, so the explicit link must remain for when it's not in scope and the lint has to be allowed for the cases where it's already in scope.